packages/marketplace/src/components/ui/submit-app-wizard/submit-app-wizard-modal.tsx

```diff
@@ -1,11 +1,61 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { Modal } from '@reapit/elements'
 import { SubmitAppWizard } from './submit-app-wizard'
 import { ModalProps } from '@/types/core'
+import { Dispatch } from 'redux'
+import { ReduxState } from '@/types/core'
+import { selectSubmitAppFormState } from '@/selector/submit-app'
+import { onFinish } from './steps/step-submit-app-success'
+import { useDispatch } from 'react-redux'
+import reduxStore from '@/core/store'
+import { submitAppSetFormState } from '@/actions/submit-app'
+
+export type OnCloseSubmitAppModalParams = {
+  getReduxState: () => ReduxState
+  dispatch: Dispatch
+  selectSubmitAppFormState
+} & Pick<ModalProps, 'afterClose'>
+
+export const customAfterClose = ({
+  dispatch,
+  afterClose,
+  getReduxState,
+  selectSubmitAppFormState,
```

Comment:
This one was for dependency injection. Was need to test selectSubmitAppFormState return for these cases: `SUCCESS` ad non `SUCCESS`).